### ISSUE TITLE
@Pure annotations in stub files appear to be ignored by nullness checker (version 1.1.3)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#101]
```
What steps will reproduce the problem?
1. Download and extract the attached test case
2. Copy the Checker Framework 1.1.2 binaries into the checkers/bin_1.1.2 
directory and the Checker Framework 1.1.3 binaries into the checkers/bin_1.1.3 
directory
3. Run 'ant clean compile.112' and note the results
4. Run 'ant clean compile.113' and note the results (compare with results from 
#3)


What is the expected output? What do you see instead?
The output using 1.1.3 should contain a single warning that looks like this:

[jsr308.javac] 
C:\Users\atomicknight\Documents\Eclipse\CheckerTest\src\com\example\Test.java:14
: warning: dereference of possibly-null reference list1.item(0)
[jsr308.javac]          System.out.println( list1.item( 0 ).toString( ) );

The actual output correctly includes the above warning, but also generates an 
additional incorrect warning:

[jsr308.javac] 
C:\Users\atomicknight\Documents\Eclipse\CheckerTest\src\com\example\Test.java:28
: warning: dereference of possibly-null reference list3.item(0)
[jsr308.javac]          System.out.println( list3.item( 0 ).getFirstChild( ) );


What version of the product are you using? On what operating system?
Version 1.1.3 on Windows 7 Professional 64-bit


Please provide any additional information below.
The attached test case tests three different scenarios to illustrate the 
differences in behavior between 1.1.2 and 1.1.3. The cases and their 
expected/actual results are as follows:

1) Annotations in source file, no @Pure annotation
- Expect a warning because the item(int) method is not annotated as @Pure
- 1.1.2 does not emit a warning (incorrect)
- 1.1.3 does emit a warning (correct)
2) Annotations in source file, @Pure annotation present
- Expect no warning
- 1.1.2 does not emit a warning (correct)
- 1.1.3 does not emit a warning (correct)
3) Annotations in stub file, @Pure annotation present
- Expect no warning
- 1.1.2 does not emit a warning (correct)
- 1.1.3 does emit a warning (incorrect)

I also tested case #3 without the @Pure annotation in the stub file, which 
resulted in no warning in 1.1.2 (incorrect) and a warning in 1.1.3 (correct).

Based on these four scenarios, it appears that 1.1.2 assumes that methods are 
pure by default (incorrect) whereas 1.1.3 correctly reads the @Pure annotation, 
but only from source files.
```

Original issue reported on code.google.com by `atomknig...@gmail.com` on 4 Jul 2011 at 8:05

Attachments:
- [CheckerTest.zip](https://storage.googleapis.com/google-code-attachments/checker-framework/issue-101/comment-0/CheckerTest.zip)
